code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -0,0 +1,360 @@
+	radio_channel = "Service" //Doesn't even use the radio anyway.
+	bot_type = HONK_BOT
+	model = "Honkbot"
+	bot_core_type = /obj/machinery/bot_core/honkbot
+	window_id = "autohonk"
+	window_name = "Honkomatic Bike Horn Unit v1.0.7"
+	data_hud_type = DATA_HUD_SECURITY_BASIC // show jobs
+
+	var/honksound = 'sound/items/bikehorn.ogg' //customizable sound
+	var/spam_flag = FALSE
+	var/cooldowntime = 30
+	var/cooldowntimehorn = 10
+	var/mob/living/carbon/target
+	var/oldtarget_name
+	var/target_lastloc = FALSE	//Loc of target when arrested.
+	var/last_found = FALSE	//There's a delay
+	var/threatlevel = FALSE
+	var/declare_arrests = FALSE // speak, you shall not, unless to Honk
+	var/idcheck = TRUE
+	var/fcheck = TRUE
+	var/check_records = TRUE
+	var/arrest_type = FALSE
+	var/weaponscheck = TRUE
+
+/mob/living/simple_animal/bot/honkbot/Initialize()
+	. = ..()
+	icon_state = "honkbot[on]"
+	auto_patrol = TRUE
+	var/datum/job/clown/J = new/datum/job/clown
+	access_card.access += J.get_access()
+	prev_access = access_card.access
+
+/mob/living/simple_animal/bot/honkbot/proc/spam_flag_false() //used for addtimer
+	spam_flag = FALSE
+
+/mob/living/simple_animal/bot/honkbot/proc/blink_end() //used for addtimer
+	icon_state = "honkbot[on]"
+
+/mob/living/simple_animal/bot/honkbot/proc/sensor_blink()
+	icon_state = "honkbot-c"
+	addtimer(CALLBACK(src, .proc/blink_end), 5)
+
+//honkbots react with sounds.
+/mob/living/simple_animal/bot/honkbot/proc/react_ping()
+	playsound(src, 'sound/machines/ping.ogg', 50, 1, -1) //the first sound upon creation!
+	spam_flag = TRUE
+	sensor_blink()
+	addtimer(CALLBACK(src, .proc/spam_flag_false), 18) // calibrates before starting the honk
+
+/mob/living/simple_animal/bot/honkbot/proc/react_buzz()
+	playsound(src, 'sound/machines/buzz-sigh.ogg', 50, 1, -1)
+	sensor_blink()
+
+/mob/living/simple_animal/bot/honkbot/bot_reset()
+	..()
+	target = null
+	oldtarget_name = null
+	anchored = FALSE
+	walk_to(src,0)
+	last_found = world.time
+	spam_flag = FALSE
+
+/mob/living/simple_animal/bot/honkbot/set_custom_texts()
+
+	text_hack = "You overload [name]'s sound control system"
+	text_dehack = "You reboot [name] and restore the sound control system."
+	text_dehack_fail = "[name] refuses to accept your authority!"
+
+/mob/living/simple_animal/bot/honkbot/get_controls(mob/user)
+	var/dat
+	dat += hack(user)
+	dat += showpai(user)
+	dat += text({"
+<TT><B>Honkomatic Bike Horn Unit v1.0.7 controls</B></TT><BR><BR>
+Status: []<BR>
+Behaviour controls are [locked ? "locked" : "unlocked"]<BR>
+Maintenance panel panel is [open ? "opened" : "closed"]"},
+
+"<A href='?src=\ref[src];power=[TRUE]'>[on ? "On" : "Off"]</A>" )
+
+	if(!locked || issilicon(user) || IsAdminGhost(user))
+		dat += text({"<BR> Auto Patrol: []"},
+
+"<A href='?src=\ref[src];operation=patrol'>[auto_patrol ? "On" : "Off"]</A>" )
+	return	dat
+
+/mob/living/simple_animal/bot/honkbot/proc/retaliate(mob/living/carbon/human/H)
+	threatlevel = H.assess_threat(src)
```

Comment:
Assess threat needs flags variable not the user. Same other places where you use it.